Added azure-li-network service

schaefi opened the pull request:
Work through the networking section of the Azure Li/Vli config
file and setup the network configuration for this instance
type. This Fixes #4

jeffaco commented on azure_li_services/units/network.py:
```diff
@@ -0,0 +1,118 @@
+    Azure Li/Vli network setup
+
+    Creates network configuration files to successfully start the
+    network in the scope of an Azure Li/Vli instance
```
Note that VLI is going to be different. It's using bridged ethernet ports. I was going to address that when we got there, but if you want it sooner, I can provide that.

I'd like to focus on just LI at this point, dealing with VLI after LI is working.

schaefi commented:
yes good point and maybe there will be more differences between LI and VLI which we should take into account at an early stage. I modified the code to allow evaluating an instance type and I suggest we add the following to the yaml config:

```yaml
version: ...
instance_type: LargeInstance|VeryLargeInstance
blade:
    ...
```

I've added a constant namespace in my code and could map any namestring from the yaml config to it. Thus if you are not in favour with the names I suggested for instance_type we could also change it to whatever fits best for you

So far the network service only runs something for InstanceType.li and will raise on anything else. We could extend the service better now, once VLI will come along

rjschwei commented on azure_li_services/network.py:
```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2017 SUSE Linux GmbH.  All rights reserved.
+#
```
It's May 2018, I know it is hard to accept that time marches on, but it does .....

schaefi replied:
:) I'll change it

rjschwei commented on azure_li_services/network.py:
```diff
@@ -0,0 +1,131 @@
+from azure_li_services.exceptions import AzureLiNetworkConfigDataException
+
+
+class AzureLiNetworkSetup(object):
```
Using Li in names that covers implementation for Li and VLi may lead to confusion at some point going forward when we need methods that are specific to one or the other type. It may be advantageous to use a generic name for such cases, maybe "AzureHosted", "AzureSAPEnv", "AzureMetal", "AzurePhys'

schaefi replied:
yes agreed, I like AzureHosted, will change it

rjschwei commented on azure_li_services/units/network.py:
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2017 SUSE Linux GmbH.  All rights reserved.
+#
```
2018

schaefi commented:
> I've done so...

Great :+1: thanks

I'll go ahead then, merge this one, prepare the image and build a new one. I'll send out an e-mail once I'm done such that @rjschwei can transfer the image for your next test round
